pkg/i2gw/providers/common/converter.go

```diff
@@ -319,7 +319,23 @@ func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv
 	var errors field.ErrorList
 	var backendRefs []gatewayv1beta1.HTTPBackendRef
 
+	// track unique backends to avoid duplicates
+	uniqueBackends := make(map[string]interface{})
+
 	for i, path := range paths {
+		if path.path.Backend.Service != nil {
+
+			backend := fmt.Sprintf("%s/%d", path.path.Backend.Service.Name, path.path.Backend.Service.Port.Number)
+
```

Comment:
Just to be bulletproof for invalid user input, I think we may want to check if `path.path.Backend.Service.Port` isn't nil

Reply:
Added the check.